### ISSUE TITLE
Compile Faust to AssemblyScript? [Feature request]

### ISSUE THREAD

**Issue by Catsvilles** [grame-cncm/faust#674]
Hi, I saw that Faust can compile to Rust, so I thought compiling to AssemblyScript also would be possible?There is already quite some work done in area of audio processing with AS, and AS is a perfect language to write audio plugins and applications for those who come from Javascript area, so I believe using them together with Faust would be just amazing :)
I personally would love to use some Faust plugins inside of my AssemblyScript app.

**Comment by sletz** [comment]
Faust also compiles wot WebAssembly. Is WebAssembly code usable in AS ?

**Comment by Catsvilles** [comment]
@sletz That was my first idea too but seems that unfortunately we cannot call WASM from AS at the moment.

**Comment by sletz** [comment]
Do you have any example of audio code or project written using AS ?

**Comment by sletz** [comment]
Great ! I guess we should invite @petersalomonsen in this discussion or continue ii on the AS Discord channel ?

**Comment by petersalomonsen** [comment]
yep, here is one example of AssemblyScript for generating real-time rendered instruments. 
AssemblyScript for the synthesis on the right side, and on the left side the music sequence is written in Javascript. It's all compiled on the fly in the browser when you press the save button.

https://petersalomonsen.com/webassemblymusic/livecodev2/?gist=7868e46c3ba499612326efbf157b5f05

**Comment by sletz** [comment]
So do we continue the discussion on AS discord channel?
